zebra/rt_netlink.c

```diff
@@ -1641,13 +1665,30 @@ static int netlink_route_multipath(int cmd, struct zebra_dplane_ctx *ctx)
 	}
 
 	if (kernel_nexthops_supported()) {
+		/* Kernel supports nexthop objects */
 		if (IS_ZEBRA_DEBUG_KERNEL)
 			zlog_debug(
 				"netlink_route_multipath(): %pFX nhg_id is %u",
 				p, dplane_ctx_get_nhe_id(ctx));
-		/* Kernel supports nexthop objects */
 		addattr32(&req.n, sizeof(req), RTA_NH_ID,
 			  dplane_ctx_get_nhe_id(ctx));
+
+		/* Have to determine src still */
+		for (ALL_NEXTHOPS_PTR(dplane_ctx_get_ng(ctx), nexthop)) {
+			if (setsrc)
+				continue;
+
+			setsrc = nexthop_set_src(nexthop, family, &src);
```

Comment:
I didn't quite get the logic here - shouldn't it be "check each nexthop; if 'setsrc', then break" ?

Reply:
I got this from how it was doing it below. You are right, I looked at that code again, it continue's because its recursive and needs to build the group. We dont have to do that sense we are sending an NH_ID so it make more sense to break in this case.

Fixed.